### PR TITLE
schema: Define a relaxed schema as v1 and move current schema to v2

### PR DESCRIPTION
**Pull request by bjhargrave**
We introduce a relaxed schema as v1 which will accept all existing taxonomy files in the main branch of the taxonomy repo. The current v1 schema is renamed to v2 and will be used for future contributions to the taxonomy repo.

The check_yaml script is updated to allow a schema version to be specified. When the schema-version option is not specified, the script will use the latest schema version from the schema base. If the schema-version is specified as a non-positive integer, then the script will use the version key from the YAML document for the schema version to use.

The script is also changed to only check for attribution.txt when the schema version is higher than 1.

See https://github.com/instructlab/instructlab/issues/989

Depends on https://github.com/instructlab/schema/pull/11

### CODE REVIEW

**Review by jjasghar — APPROVED** [review]

File: .github/scripts/check-yaml.py
Comment (jjasghar) [inline .github/scripts/check-yaml.py:137]: Ah there we go

**Comment by bjhargrave** [comment]
Before merging this PR, we should probably wait until the cli containing https://github.com/instructlab/instructlab/pull/1009 is published. If we merge before then, then we will require `version: 2` while the cli will not accept `version: 2`.

**Comment by bjhargrave** [comment]
> Before merging this PR, we should probably wait until the cli containing [instructlab/instructlab#1009](https://github.com/instructlab/instructlab/pull/1009) is published.

This happened this afternoon. `ilab` has released [`v0.14.0`](https://github.com/instructlab/instructlab/releases/tag/v0.14.0) (aka [`stable`](https://github.com/instructlab/instructlab/releases/tag/stable)) which includes the `version: 2` schema support.

So we can merge this PR when the backend is ready.

**Pull request merged**
